### ISSUE TITLE
Create functionality for TStudio Import/merge capability

### ISSUE THREAD

**Issue by RRohrssen** [unfoldingWord-dev/ts-desktop#525]

**Comment by cckozie** [comment]
I have seen the merge process both work and not work. Here is an example of an instance when it does not appear to work properly.
In this project:
[en-x-demo1_php_text_ulb.tstudio.zip](https://github.com/unfoldingWord-dev/ts-desktop/files/271206/en-x-demo1_php_text_ulb.tstudio.zip)
Chapter 1, chunk 1 looks like this:
![php1-1](https://cloud.githubusercontent.com/assets/8171759/15371691/57d6325c-1d02-11e6-9933-8c4d2c1af6f2.jpg)
In this project:
[en-x-demo1_php_text_ulb.2016-05-18_10.34.31.tstudio.zip](https://github.com/unfoldingWord-dev/ts-desktop/files/271216/en-x-demo1_php_text_ulb.2016-05-18_10.34.31.tstudio.zip)
Chapter 1, chunk 1 looks like this:
![php1-2](https://cloud.githubusercontent.com/assets/8171759/15371827/06ebcfa4-1d03-11e6-8c37-90442c5d3169.jpg)
If you start with the first project above and then import the second one, chapter 1, chuck one looks like this, although I expected to see two different versions in the chunk.
![php1-3](https://cloud.githubusercontent.com/assets/8171759/15371855/275115ce-1d03-11e6-844f-6ca6c1b85c04.jpg)

**Comment by neutrinog** [comment]
@cckozie git is smart and it will automatically merge things together when it can. The text in both of the projects you've shown are identical therefore the result you are seeing is correct.

**Comment by neutrinog** [comment]
it would be better to test with different text in the chunks to ensure a merge conflict will occur.

**Comment by cckozie** [comment]
The text is similar but different ("to all **who** are set apart" vs. "to all **the ones** set apart").

**Comment by cckozie** [comment]
This is still failing in the 8.0.3 build.

**Comment by cdwhitfield73** [comment]
Cannot duplicate this error.  Using the exact same text as above, when I merge I get both versions in chunk 1

**Comment by neutrinog** [comment]
@cdwhitfield73 @cckozie the files used in this test are incorrect. Please see https://github.com/unfoldingWord-dev/ts-android/issues/1518. Note also however the separate error importing these files (from android) into desktop.

**Comment by cckozie** [comment]
I created these new files with the same data in chunks 1 and 2 as above using this process:
[en-x-demo1_php_text_regA.tstudio.zip](https://github.com/unfoldingWord-dev/ts-desktop/files/284447/en-x-demo1_php_text_regA.tstudio.zip)
[en-x-demo1_php_text_regB.tstudio.zip](https://github.com/unfoldingWord-dev/ts-desktop/files/284450/en-x-demo1_php_text_regB.tstudio.zip
1. created project ...A
2. backed up project ...A
3. deleted project ...A
4. created project ...B
5. imported project ...A --- Results were as they should be, two versions of chunk 1 now in the project
6. deleted project
7. restored project ...A
8. imported project...B --- Results were incorrect, only one version of chunk 1 now in project
9. deleted project
10. restored project ...B
11. imported project ...A --- Results were correct, two versions of chunk 1 now in project
12. deleted project and repeated steps 7 and 8 with the same result as 8
13. deleted project and repeated steps 10 and 11. This time the import failed with this message:
![image](https://cloud.githubusercontent.com/assets/8171759/15576200/5d434c82-231c-11e6-8deb-bef9d535288d.png)

**Comment by cckozie** [comment]
I repeated the process above on Win10 with the opposite result. 
If I import the first file created into the second one, I end up with only one version of the text in chunk 1.
If I import the second file create into the first one, I end up with two versions of the text in chunk 1.

**Comment by cdwhitfield73** [comment]
In the 13 step process above, there is something missing.  Before importing B in step 8, it doesn't say where or when you backed it up or created it.  If you made a backup between steps 5 and 6, this backup would contain the git history of both projects, and would explain the results of the second merge.  Since we are merging git histories, not just text, every time you perform a merge, it must be with two completely independent projects in order to get the results you want.  By deleting and restoring and merging the same two projects over and over, you leave room for unexpected results based on not really knowing where the git history is on each project.  In my mind, for the user story that we should theoretically have on this, the fact that the results were as expected after step 5 is a Pass.  If we need more advanced code to handle more complex situations, that should be a new issue, but I don't see scenarios where actual users are going to try to perform these steps you presented past step 5.  What you should test is after step 5, try to import a third independent project and see what happens.  This is the scenario that will happen on the field.

**Comment by cdwhitfield73** [comment]
So I figured out why the conflict error messages were coming up.  I could not duplicate the issue while using my own files, but when I imported the 2 files Koz attached above, I discovered that one of them (B) had somehow been archived with uncommitted changes in it.  I couldn't figure out how he could have done that when the project gets committed every time you leave the home page.  Then it dawned on me that you can create a backup of the project while the project is still open without going to the home page, in which case, any changes you have made since the last auto commit on the timer would not be committed.  I will address that in a couple ways.  First of all, I will commit projects before backing up.  As an added safety net, I think I will commit all projects right before trying to merge them as well.  This should totally eliminate the possibility of getting the "conflicts prevent checkout" error.
